lib/connect/index.js

```diff
@@ -105,8 +105,13 @@ function connect (brokerUrl, opts) {
   }
 
   if (!protocols[opts.protocol]) {
-    opts.protocol = protocolList.filter(function (key) {
-      return typeof protocols[key] === 'function'
+    const isSecure = (['mqtts', 'wss'].indexOf(opts.protocol) !== -1)
+    opts.protocol = protocolList.filter(function (key, index) {
+      if (isSecure && index % 2 === 0) {
```

Comment:
One of my goals was to not disrupt the existing code - might have taken it too far, usually `%`ing is a red flag for that :) . @mcollina requested in #392 that we not use too many ifs which is why I went with my solution. So yes we could do this:

```js
// Define secureProtocolList and protocolList above...
  if (!protocols[opts.protocol]) {
    const isSecure = secureProtocolList.indexOf(opts.protocol) !== -1
    if (isSecure) {
      opts.protocol = secureProtocolList.filter(function (key, index) {
        return (typeof protocols[key] === 'function')
      })[0]
    } else {
      opts.protocol = protocolList.filter(function (key, index) {
        return (typeof protocols[key] === 'function')
      })[0]
    }
  }
```

Issues: we have a `protocols` object a `protocolList` array and a `secureProtocolList` array. Two of which are only used in this one block of code.

or I just pushed a solution that instead makes the dependency much more obvious and removes the `protocolList`. This block was the only part that used the `protocolList` so it makes sense to just include it directly and no one will go making changes without looking inside the filter function.

There was always a requirement on the order of the array so I don't think it's an issue that it has to go insecure, secure, since before it had to be in that order too (otherwise I would have just changed the order). The problem is with the list being defined so far away it could have been altered too easily. My last commit fixes that :)